Sources/Deprecated.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Deprecated.swift
+//  APIKit
+//
+//  Created by Yosuke Ishikawa on 9/19/16.
+//  Copyright © 2016 Yosuke Ishikawa. All rights reserved.
+//
+
+import Result
+
+// MARK: - Protocols
+
+@available(*, deprecated, renamed: "Request")
```

Comment:
Should this be `deprecated` or `unavailable`? If we choose the former, we can't remove these migration assistance until we get a next major version due to Semantic Versioning conformance. It might makes sense to let this `deprecated` in `3.0.0-beta.2` prerelease and make it `unavailable` in final `3.0.0` release.